src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -275,19 +283,23 @@ namespace levin
            algorithm changes or the locking strategy within the levin config
            class changes. */
 
-        std::vector<boost::uuids::uuid> connections;
+        std::vector<detail::p2p_context> connections;
         connections.reserve(connection_id_reserve_size);
         zone_->p2p->foreach_connection([this, &connections] (detail::p2p_context& context) {
           /* Only send to outgoing connections when "flooding" over i2p/tor.
              Otherwise this makes the tx linkable to a hidden service address,
              making things linkable across connections. */
           if (this->source_ != context.m_connection_id && (this->zone_->is_public || !context.m_is_income))
-            connections.emplace_back(context.m_connection_id);
+            connections.emplace_back(context);
           return true;
         });
 
-        for (const boost::uuids::uuid& connection : connections)
-          zone_->p2p->send(message_.clone(), connection);
+        std::string category = "command-" + std::to_string(get_command_from_message(message_));
```

Comment:
The result of `get_command_from_message` can be given directly to `on_levin_traffic` now.